src/ucp/wireup/wireup_cm.c

```diff
@@ -364,11 +371,21 @@ static unsigned ucp_ep_cm_remote_disconnect_progress(void *arg)
     ucs_assert(ucp_ep->flags & UCP_EP_FLAG_LOCAL_CONNECTED);
     if (ucs_test_all_flags(ucp_ep->flags, UCP_EP_FLAG_CLOSED |
                                           UCP_EP_FLAG_CLOSE_REQ_VALID)) {
+        if (!UCS_PTR_IS_PTR(ucp_ep_ext_gen(ucp_ep)->close_req.req)) {
```

Comment:
this should be ucs_assert()
we can do 
```
if (CLOSED) {
  if (CLOSE_REQ_VALID) {
     assert(PTR_IS_PTR...)
     complete_send
  }
  return 1
}
```

Reply:
I don't think segfault is a good idea for client/server app even if there is no memory

Reply:
I don't think segfault is a good idea for client/server app even if there is no memory

Reply:
Why segfault? I thought this function is called only if close_nb allocated a request?

Reply:
If local side is initiator of close protocol it calls uct_ep_disconnect if flush is completed in place, remote side will respond and local side have to handle this in disconnect callback. IMO better to print error vs switch ep to wrong close protocol state not setting request (or err ptr) at all

Reply:
Still dlnt understand, if clise request is not valid, how could it be added to ep in the first place?

Reply:
See https://github.com/openucx/ucx/pull/4445#discussion_r357607189 again, by ucp_ep_set_close_request on initiator side if flush is completed immediately

Reply:
so i think we should not set close request **in case request allocation failed**. we should treat it as if close_nb was not called, e.g just send disconnect reply (if not sent already)

Reply:
I'm afraid we will have the ep in wrong state which may cause UB of protocol. I'll double check how it may look like now and how is it sensitive for modifications.

Reply:
what is "UB"? 
if request was not allocated then ucp_close_nb would not have any side-effect if close request is not set on the ep (flush is not considered side effect because it could happen anytime regardless of close)